packages/quri_parts/jijmodeling_transpiler_quantum/quri_parts/qaoa/to_qaoa_quri.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+import qiskit as qk
+import jijmodeling as jm
+import jijmodeling.transpiler as jmt
+from jijmodeling_transpiler_quantum.core import qubo_to_ising
+from .ising_hamiltonian_quri import to_ising_operator_from_qubo_quri
+from quri_parts.circuit import LinearMappedUnboundParametricQuantumCircuit
+from quri_parts.core.operator import Operator
+from math import pi
+
+
+class QuriQAOAAnsatzBuilder:
+    def __init__(
+        self,
+        pubo_builder: jmt.core.pubo.PuboBuilder,
+        num_vars: int,
+        compiled_instance: jmt.core.CompiledInstance,
+    ):
+        self.pubo_builder = pubo_builder
+        self.num_vars = num_vars
+        self.compiled_instance = compiled_instance
+
+    @property
+    def var_map(self) -> dict[str, tuple[int, ...]]:
+        return self.compiled_instance.var_map.var_map
+
+    def get_hamiltonian(
+        self,
+        multipliers: dict = None,
+        detail_parameters: dict = None,
+    ) -> tuple[Operator, float]:
+        qubo, constant = self.pubo_builder.get_qubo_dict(
+            multipliers=multipliers, detail_parameters=detail_parameters
+        )
+        ising_operator, ising_const = to_ising_operator_from_qubo_quri(
+            qubo, self.num_vars
+        )
+        return ising_operator, ising_const + constant
+
+    def get_qaoa_ansatz(
+        self,
+        p: int,
+        multipliers: dict = None,
+        detail_parameters: dict = None,
+    ) -> tuple[UnboundParametricQuantumCircuit, operator, float]:
+        ising_operator, constant = self.get_hamiltonian(
+            multipliers=multipliers, detail_parameters=detail_parameters
+        )
+        pauli_terms = []
+        coeff_terms = []
+        pauli_z = []
+        pauli_zz = []
+        pauli_z_coeff = []
+        pauli_zz_coeff = []
+
+        for i in range(ising_operator.n_terms - 1):
+            keys = list(ising_operator.keys())[i]
+            pauli_terms.append(keys.index_and_pauli_id_list[0])
+            items = list(ising_operator.items())[i]
```

Comment:
If you create same list, don't create it in the loop, please do it outside of the loop.